velox/functions/prestosql/aggregates/tests/MapAggTest.cpp

```diff
@@ -482,5 +482,74 @@ TEST_F(MapAggTest, rowCheckNull) {
       "ROW comparison not supported for values that contain nulls");
 }
 
+TEST_F(MapAggTest, nans) {
+  // Verify that NaNs with different binary representations are considered equal
+  // and deduplicated when used as keys in the output map.
+  static const auto kNaN = std::numeric_limits<double>::quiet_NaN();
+  static const auto kSNaN = std::numeric_limits<double>::signaling_NaN();
+
+  // Global Aggregation, Primitive type
+  auto data = makeRowVector(
+      {makeFlatVector<double>({1, kNaN, kSNaN, 2, 3, kNaN, kSNaN, 3}),
+       makeFlatVector<int32_t>({1, 2, 3, 4, 5, 6, 7, 8}),
+       makeFlatVector<int32_t>({1, 1, 1, 1, 2, 2, 2, 2})});
+
+  auto expectedResult = makeRowVector({
+      makeMapVectorFromJson<double, int32_t>({
+          "{ 1: 1, 2: 4, 3: 5, NaN: 2}",
+      }),
+  });
+
+  testAggregations({data}, {}, {"map_agg(c0, c1)"}, {expectedResult});
```

Comment:
map_agg is order sensitive , thus its better to write this as "map_agg(c0, c1 order by c1 asc)" to be fully deterministic..

Reply:
adding an ordering over the value prevents testAggregations from testing out all aggregation plans with an error `Order sensitive aggregation over sorted inputs cannot be split into partial and final: map_agg(c0, c1 order by c1 asc).`

I am assuming that since the input is small and a single vector, the execution would be single threaded and be deterministic similar to other tests in this suite. We can re-visit this later if it becomes flaky at any point.